cpp/src/IceStorm/Scanner.l

```diff
@@ -1,64 +1,76 @@
-%{
-
+%top{
 //
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
 
+#include "IceUtil/ScannerConfig.h"
+#include <cstdint>
+
+}
+
+%{
+
 #include "Ice/Ice.h"
 #include "Parser.h"
 #include "Grammar.h"
 
-#if defined(_MSC_VER)
-// '<' : signed/unsigned mismatch
-#   pragma warning(disable:4018)
-// 'initializing' : conversion from '__int64' to 'int', possible loss of data
-#   pragma warning(disable:4244)
-
-//
-// '=' : conversion from 'size_t' to 'int', possible loss of data
-// The result of fread() is a size_t and gets inserted into an int
-//
-#   pragma warning(disable:4267)
-#endif
-
-#if defined(__GNUC__)
-#   pragma GCC diagnostic ignored "-Wsign-compare"
+#ifdef _MSC_VER
+#    ifdef yywrap
+#        undef yywrap
+#        define yywrap() 1
+#    endif
+#    define YY_NO_UNISTD_H
 #endif
 
 using namespace std;
 using namespace Ice;
 using namespace IceStorm;
 
-#ifdef _MSC_VER
-#   ifdef yywrap
-#      undef yywrap
-#      define yywrap() 1
-#   endif
-#   define YY_NO_UNISTD_H
-#endif
-
-#define YY_INPUT(buf, result, maxSize) parser->getInput(buf, result, maxSize)
-
 namespace IceStorm
 {
+    typedef map<string, int> StringTokenMap;
+    StringTokenMap keywordMap;
 
-typedef std::map<std::string, int> StringTokenMap;
-static StringTokenMap keywordMap;
-
-void initScanner();
-
+    void initScanner();
 }
-#define         YY_USER_INIT initScanner();
+
+// Override some of the functions flex auto-generates with our own implementations.
+#define YY_USER_INIT initScanner();
+#define YY_INPUT(buf, result, maxSize) parser->getInput(buf, result, maxSize)
 
 %}
 
+  /* Instructs flex to not suppress any warnings when generating the scanner. */
```

Comment:
Can we use our regular comment style, `//` for non-doc comments? The indentation of the comment looks odd too.

Reply:
Unfortunately not. Flex only supports C-style comments, and this indentation is also necessary.
https://www.cs.virginia.edu/~cr4bd/flex-manual/Comments-in-the-Input.html